### ISSUE TITLE
No code actions in emacs after upgrading nixpkgs

### ISSUE THREAD

**Issue by samuelrivas** [haskell/haskell-language-server#3040]
### Your environment

Which OS do you use: Ubuntu 20.04.4 LTS. But emacs, haskell, etc come from nixpkgs (commit hashes included below)
Which LSP client (editor/plugin) do you use: emacs+lsp-mode
Describe your project (alternative: link to the project): just ghc (no cabal, or stack)

### Steps to reproduce

I can reproduce it with a simple haskell file like this:

```haskell
{-# OPTIONS_GHC -Wall #-}

import           Data.Monoid (mempty)

test :: ()
test =  ()
```

### Expected behaviour
When placing the cursor on the import line, I expect to see a warning and a list of code actions to remove the unused imports.

![working](https://user-images.githubusercontent.com/107703/178668995-aaa4493f-7738-47c9-a2ff-2a3fdc0afc5c.png)

### Actual behaviour

I just see the warning, but no code actions available. Running `lsp-ui-sideline-apply-code-actions` shows `"No code actions on the current line"` on the minibuffer.

Other features of lsp seem to be working as expected. I haven't spotted any other issue. Code actions don't work on any other cases that I tested.

![not-working](https://user-images.githubusercontent.com/107703/178669283-91f8516f-16ef-417a-9f3b-b1f891464327.png)

### Include debug information

I see the effect after upgrading nixpkgs, from commit `aa2f845096f72dde4ad0c168eeec387cbd2eae04` to `d423c0ce15adf573c2045bef5c921b958b674bb4`.

I don't see any obvious errors in emacs, but there is a formatting error in `*Messages*` that could be related
```
LSP :: Connected to [lsp-haskell:332529/starting].
Starting new Ispell process /nix/store/hj082w7fj7pwyscy6s05aghkqsnncq1v-emacs-packages-deps/bin/aspell with default dictionary...done
LSP :: lsp-haskell:332529 initialized successfully in folders: (/home/samuel.rivas/src/monorepos/sams-monorepo/src/haskell/monad-emit/src/src)
Invalid face reference: lsp-flycheck-warning-unnecessary [42 times]
previous-line: Beginning of buffer
Mark set
Invalid face reference: lsp-flycheck-warning-unnecessary [8 times]
```

Running `haskell-language-server --probe-tools` returns:

  - not working case
```
2022-07-12T09:12:58.450112Z | Info | No log file specified; using stderr.
haskell-language-server version: 1.7.0.0 (GHC: 9.0.2) (PATH: /nix/store/pwyfrf1sin06mdv7a0wjgsfrnil1h57p-haskell-language-server-1.7.0.0/bin/haskell-language-server)
Tool versions found on the $PATH
cabal:          Not found
stack:          Not found
ghc:            9.0.2
```
  - working case
```
haskell-language-server version: 1.4.0.0 (GHC: 8.10.7) (PATH: /nix/store/k3aqqp8f5ybs22a029xp0p87slnmn87r-haskell-language-server-1.4.0.0/bin/haskell-language-server)
Tool versions found on the $PATH
cabal:          Not found
stack:          Not found
ghc:            8.10.7
```

I can also reproduce this error with ghc 8.10.7 by pointing my new nixpkgs version to a haskellPackages from the previous hash. in this case `--probe-tools` returns

 - not working case
```
haskell-language-server --probe-too
haskell-language-server version: 1.4.0.0 (GHC: 8.10.7) (PATH: /nix/store/k3aqqp8f5ybs22a029xp0p87slnmn87r-haskell-language-server-1.4.0.0/bin/haskell-language-server)
![not-working](https://user-images.githubusercontent.com/107703/178668517-a12c7e6c-43ab-452d-9308-742b1399e949.png)
![working](https://user-images.githubusercontent.com/107703/178668540-9e6a8e7d-4030-4cd3-a44b-939f2cc94fc3.png)

Tool versions found on the $PATH
cabal:          Not found
stack:          Not found
ghc:            8.10.7
```

In short. GHC 8.10.7, haskell language server 1.4.0.0 works in my old nixpkgs version. The same versions of GHC and haskell language server fail on my new nixpkgs version. GHC 9.0.2 and haskell language server 1.7.0.0 still fail. So it does not seem to be related to either ghc or haskell language server. The emacs version is the same in both cases (28.1), but the Emacs packages change.

Any suggestions on how to further debug this?

**Comment by samuelrivas** [comment]
Update: Another thing that does not work on my new configuration is the `lsp-ui` popups with type and documentation info.

Working case:
![working-ui](https://user-images.githubusercontent.com/107703/178673874-1e45848e-abd1-400c-beb5-d6f2feef4d97.png)

Not working case:
![not-working-ui](https://user-images.githubusercontent.com/107703/178673951-2103d5cb-40af-4ac8-a52d-52f92d7ae8b4.png)

**Comment by Ailrun** [comment]
I guess it might be to do with the version of lsp-mode? I experienced similar issues in other lsp servers as well in one of my machine.

**Comment by michaelpj** [comment]
I think you're going to have to go look through the logs. Have a look at the troubleshooting documentation.

**Comment by samuelrivas** [comment]
Thanks, I'll take a look to the troubleshooting docs and see what I can find.

**Comment by samuelrivas** [comment]
Looking at the lsp log in emacs, I think the language server may be working as expected. I see the code actions there:

```
[Trace - 07:16:29 AM] Sending request 'textDocument/documentHighlight - (29)'.
Params: {
  "textDocument": {
    "uri": "file:///home/samuel.rivas/src/monorepos/sams-monorepo/src/haskell/monad-emit/src/src/minimal.hs"
  },
  "position": {
    "line": 2,
    "character": 0
  }
}


[Trace - 07:16:29 AM] Received response 'textDocument/documentHighlight - (29)' in 1ms.
Result: []


[Trace - 07:16:29 AM] Received response 'textDocument/codeAction - (28)' in 22ms.
Result: [
  {
    "edit": {
      "changes": {
        "file:///home/samuel.rivas/src/monorepos/sams-monorepo/src/haskell/monad-emit/src/src/minimal.hs": [
          {
            "range": {
              "start": {
                "line": 1,
                "character": 0
              },
              "end": {
                "line": 1,
                "character": 0
              }
            },
            "newText": "{-# OPTIONS_GHC -Wno-unused-imports #-}\n"
          }
        ]
      }
    },
    "kind": "quickfix",
    "diagnostics": [],
    "title": "Disable \"unused-imports\" warnings"
  },
  {
    "edit": {
      "changes": {
        "file:///home/samuel.rivas/src/monorepos/sams-monorepo/src/haskell/monad-emit/src/src/minimal.hs": [
          {
            "range": {
              "start": {
                "line": 2,
                "character": 0
              },
              "end": {
                "line": 3,
                "character": 0
              }
            },
            "newText": ""
          }
        ]
      }
    },
    "kind": "quickfix",
    "diagnostics": [
      {
        "severity": 2,
        "range": {
          "start": {
            "line": 2,
            "character": 0
          },
          "end": {
            "line": 2,
            "character": 37
          }
        },
        "code": "-Wunused-imports",
        "source": "typecheck",
        "message": "The import of ‘Data.Monoid’ is redundant\n  except perhaps to import instances from ‘Data.Monoid’\nTo import instances alone, use: import Data.Monoid()",
        "tags": [
          1
        ]
      }
    ],
    "title": "Remove import"
  },
  {
    "edit": {
      "changes": {
        "file:///home/samuel.rivas/src/monorepos/sams-monorepo/src/haskell/monad-emit/src/src/minimal.hs": [
          {
            "range": {
              "start": {
                "line": 2,
                "character": 0
              },
              "end": {
                "line": 3,
                "character": 0
              }
            },
            "newText": ""
          }
        ]
      }
    },
    "kind": "quickfix",
    "title": "Remove all redundant imports"
  }
]
```
So I think @Ailrun is right about lsp-mode being at fault.

I'll check with other versions

**Comment by jeslie0** [comment]
You could try seeing if code actions appear in Eglot - If they do then the fault is with lsp-mode.

**Comment by samuelrivas** [comment]
I made it work with language server 1.7.0.0 using the previous version of all emacs packages, so it is almost definitely not an issue of language server but a problem in some emacs package.

Feel free to close this issue if you prefer, otherwise I'll update when I identify which package is failing.

Thanks for your help!

**Comment by samuelrivas** [comment]
It seems that is [lsp-ui](https://github.com/emacs-lsp/lsp-ui) what breaks it. Without changing anything else, I can reproduce but with commit `eba9c4eaa255a14e2facd658e7122674c05390f8` (current head of lsp-ui) , but code actions work with commit 
`dd4c181a22d19a28236c442cf6c9cd4bbd6d85f8` (the version I had before upgrading).

I'll move this issue there

**Comment by samuelrivas** [comment]
I found the issue. Sideline actions and doc popups need to be explicitly enabled now for `lsp-ui`

Thanks for your help here!
